File: android/app/src/main/kotlin/com/typography/animations/entrance/Spin3D.kt
package com.typography.animations.entrance

import com.typography.AnimatedTextState

fun evaluateSpin3D(t: Float): AnimatedTextState {
    val rotationX = -180.0f * (1.0f - t)
    val rotationY = -180.0f * (1.0f - t)
    val scale = 0.2f + 0.8f * t
    return AnimatedTextState(
        opacity = t.coerceIn(0f, 1f),
        rotationX = rotationX,
        rotationY = rotationY,
        scale = scale
    )
}

File: lib/utils/animations/entrance/spin3d.dart
import '../base/animated_text_state.dart';

AnimatedTextState evaluateSpin3D(double t) {
  final rotationX = -180.0 * (1.0 - t);
  final rotationY = -180.0 * (1.0 - t);
  final scale = 0.2 + 0.8 * t;
  return AnimatedTextState(
    opacity: t.clamp(0.0, 1.0),
    rotationX: rotationX,
    rotationY: rotationY,
    scale: scale,
  );
}
